### PR TITLE
Clean up C build dependencies (and bump Qiskit)

### PR DESCRIPTION
**Pull request by wshanks**
* Remove muparserx which is not needed as of version 0.13
* Ignore run exports from spdlog. Only its headers are used at build time.
* Make numpy a run dependency only. Qiskit Aer interfaces with numpy through the pybind11 headers and does not link directly to numpy.
* Make the nlohmann_json bounds consistent across environemnts instead of special for osx arm64.
* Increase qiskit minimum version

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by wshanks** [comment]
@leofang I am curious what you think about these changes. The warnings in the build log about DSO's and exports made me look loser at the linking of the C extension. I noticed that muparserx was dropped in the last release, so that was an easy change (as was realizing that I missed bumping the minimum qiskit version).

I also noticed that the extension does not link against the spdlog library. I think it only uses the headers at build time and doesn't need spdlog at run time.

I also noticed that the extension does not link against numpy and looking at the [pybind11 documentation](https://pybind11.readthedocs.io/en/stable/advanced/pycpp/numpy.html#arrays) I read that pybind has its own numpy headers and does not need numpy at build time. I am a little concerned about numpy 2 based on [this issue](https://github.com/pybind/pybind11/issues/5009), but I think the pybind11 feedstock should set some kind of export on numpy if that is a problem rather than bounding numpy here. However, maybe just depending on numpy is more standard for getting the right run constraints any way for conda-forge?

I also bumped up the nlohmann_json version -- I just thought if we are using something different from upstream for one variant and it is working okay we might as well use it for all the variants.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,6 +23,8 @@ source:
     # This patch was accepted in https://github.com/Qiskit/qiskit-aer/pull/1940
     # and can be removed from the receipe with the 0.14 release.
     - 0001-windows-find-blas.patch
+    # muparserx usage was removed in https://github.com/Qiskit/qiskit-aer/pull/1884
+    - 0001-Remove-muparserx-from-cmake-configuration.patch
```
Comment (wshanks) [inline recipe/meta.yaml:27]: The patch is in https://github.com/Qiskit/qiskit-aer/pull/2045. I hadn't opened that when I pushed here or I would have put the link a comment. I don't want to churn through the whole CI for that, but I will put it in if I do another push.

**Comment by leofang** [comment]
Thanks, Will, a bit overwhelmed this week, will try to take a close look asap.

**Comment by wshanks** [comment]
Thanks! No rush. The changes seem to pass the tests at least.

**Review by leofang — APPROVED** [review]
LGTM, sorry for delay

**Comment by leofang** [comment]
@wshanks feel free to resolve the conflicts and merge

**Comment by wshanks** [comment]
@conda-forge-admin, please rerender

**Comment by wshanks** [comment]
When I rerendered locally, all the CUDA 12 builds got removed, so I am trying rerendering with the bot here.

**Comment by wshanks** [comment]
@leofang It looks like it also removed the CUDA 12 build here and in #34. Any idea why or how to avoid that?

**Pull request merged**